### ISSUE TITLE
Content-Type header is getting overridden

### ISSUE THREAD

**Issue by ourdex86** [amitshekhariitbhu/Fast-Android-Networking#436]
I need to pass 
**Content-Type : application/x-www-form-urlencoded; charset=UTF-8**

as the header. But when I try the following  way

```
        AndroidNetworking
            .post(Url.BASE_URL + requestPath)
            .addHeaders("Content-Type", "application/x-www-form-urlencoded; charset=utf-8")
            .setContentType("application/x-www-form-urlencoded; charset=utf-8")
            .addBodyParameter(requestPath.split("/").last(), jsonString)
            .setPriority(Priority.MEDIUM)
            .build()
            .getAsJSONObject(jsonObjectRequestListener)
```

its setting the header as **Content-Type: application/x-www-form-urlencoded**

and **charset=UTF-8** is getting removed. How to forcefully add this complete thing as header ?

**Comment by amitshekhariitbhu** [comment]
Use any of these from the following in place of `addBodyParameter`:
* addApplicationJsonBody
* addJSONObjectBody
* addJSONArrayBody
* addStringBody

**Comment by ourdex86** [comment]
But this is form url encoded request. Thats why I am using using **addBodyParameter**

**Comment by amitshekhariitbhu** [comment]
Then use `addUrlEncodeFormBodyParameter`

**Comment by ourdex86** [comment]
Yes, I tried that. Still that won't append **charset=utf-8** in the header. Even after setting that forcefully, its getting removed.

**Comment by Elswah** [comment]
AndroidNetworking.post("https://fierce-cove-29863.herokuapp.com/createAnUser")
                 .addBodyParameter("firstname", "Amit")
                 .addBodyParameter("lastname", "Shekhar")
                 .setContentType("application/json; charset=utf-8") // custom ContentType
                 .setPriority(Priority.MEDIUM)
                 .build()
                 .getAsJSONArray(new JSONArrayRequestListener() {
                    @Override
                    public void onResponse(JSONArray response) {
                      // do anything with response
                    }
                    @Override
                    public void onError(ANError error) {
                      // handle error
                    }
                });

this is the only way to send  content type when iam trying to send  application/x-www-form-urlencoded
in addheader method not working i want to send it in add header how to do that

**Comment by ourdex86** [comment]
> ```
> AndroidNetworking.post("https://fierce-cove-29863.herokuapp.com/createAnUser")
>                  .addBodyParameter("firstname", "Amit")
>                  .addBodyParameter("lastname", "Shekhar")
>                  .setContentType("application/json; charset=utf-8") // custom ContentType
>                  .setPriority(Priority.MEDIUM)
>                  .build()
>                  .getAsJSONArray(new JSONArrayRequestListener() {
>                     @Override
>                     public void onResponse(JSONArray response) {
>                       // do anything with response
>                     }
>                     @Override
>                     public void onError(ANError error) {
>                       // handle error
>                     }
>                 });
> ```
> 
> this is the only way to send content type when iam trying to send application/x-www-form-urlencoded
> in addheader method not working i want to send it in add header how to do that. i dont want to use method setContentType i want to make this in AddHeader method only.

**Comment by ourdex86** [comment]
@amitshekhariitbhu .. Sorry by mistake it was closed. Any update on this ?
